routes/deployments.py

```diff
@@ -0,0 +1,163 @@
+import json
+
+import requests
+from flask import flash
+
+import config
+import routes
+from github_api import get_default_branch, get_last_commit_sha
+from gitlab_api import (
+    get_app_interface_deploy_config_files,
+    get_app_interface_file_content,
+)
+from utils import get_app_interface_folders, save_json_data_and_return
+
+IGNORE_LIST = (
+    "floorist",
+    "-ui",
+    "access-permissions",
+    "frontend",
+    "routes",
+    "-dev",
+    "ephemeral",
+)
+
+
+def get_deployments(reload_data=None):
+    """
+    Get deployments data from the file or download them.
+    """
+    if config.GITLAB_TOKEN and config.GITHUB_TOKEN:
+        if not config.DEPLOYMENTS_LIST.is_file() or reload_data:
+            get_app_interface_deployments()
+    else:
+        if not config.DEPLOYMENTS_LIST.is_file():
+            return {}
+
+    with open(config.DEPLOYMENTS_LIST, mode="r", encoding="utf-8") as file:
+        return json.load(file)
+
+
+def get_app_interface_deployments():
+    """
+    Download deployments data from app-interface repository
+    (https://gitlab.cee.redhat.com/service/app-interface)
+    based on links from the Overview page.
+    """
+    services_links = routes.overview_page.get_services_links()
+    app_interface_folders = get_app_interface_folders(
+        services_links, config.APP_INTERFACE_PATTERN
+    )
+    deployments = {}
+
+    for folder, app_interface_link in app_interface_folders:
```

Comment:
Three for loops and one try/except block is kinda too much and it would be better to refactor this. But it doesn't have to be done now.